File: services/rag/app/utils/folder_path.py
"""Folder path normalization for the folder-scoped search filter.

The platform's canonical folder path format is ``parent/child`` — segments
joined by ``/`` with no leading or trailing slash (see
``services/platform/convex/folders/queries.ts::buildFolderPath``). The
TypeScript twin of this helper lives at
``services/platform/convex/lib/helpers/rag_folder_path.ts`` — keep both in
sync.
"""

from __future__ import annotations

MAX_FOLDER_PATH_LENGTH = 1024

_STRIP_CHARS = " \t\r\n/"


def normalize_folder_path(value: object) -> str | None:
    """Normalize a folder path to the canonical ``parent/child`` form.

    Strips surrounding whitespace and slashes. Returns ``None`` for
    non-string, empty, or all-separator values so callers can treat the
    result as "no folder".
    """
    if not isinstance(value, str):
        return None
    normalized = value.strip(_STRIP_CHARS)
    return normalized or None

File: services/platform/convex/rag/lib/folder_path.ts
'use node';

/**
 * Folder path normalization for the folder-scoped search filter.
 *
 * The platform's canonical folder path format is `parent/child` — segments
 * joined by `/` with no leading or trailing slash. The TypeScript twin in the
 * platform lives at `services/platform/convex/lib/helpers/rag_folder_path.ts`.
 */

export const MAX_FOLDER_PATH_LENGTH = 1024;

// Mirror Python's `value.strip(" \t\r\n/")` exactly — only these characters,
// trimmed from both ends.
const STRIP_RE = /^[ \t\r\n/]+|[ \t\r\n/]+$/g;

/**
 * Normalize a folder path to the canonical `parent/child` form. Strips
 * surrounding whitespace and slashes. Returns `null` for non-string, empty,
 * or all-separator values so callers can treat the result as "no folder".
 */
export function normalizeFolderPath(value: unknown): string | null {
  if (typeof value !== 'string') {
    return null;
  }
  const normalized = value.replace(STRIP_RE, '');
  return normalized.length > 0 ? normalized : null;
}
